### PR TITLE
Add /readiness to HistoricalResource

### PR DESCRIPTION
**Pull request by leventov**
Add an endpoint that returns local cache load status as HTTP response code: 200 or 503.

### CODE REVIEW

File: server/src/main/java/io/druid/server/http/HistoricalResource.java [hunk 1]
```diff
@@ -52,4 +52,16 @@ public Response getLoadStatus()
...
     return Response.ok(ImmutableMap.of("cacheInitialized", coordinator.isStarted())).build();
   }
+
+  @GET
+  @Path("/loadStatusCode")
+  @Produces(MediaType.APPLICATION_JSON)
+  public Response getLoadStatusCode()
+  {
+    if (coordinator.isStarted()) {
+      return Response.ok(ImmutableMap.of()).build();
```
Comment (xanec) [inline server/src/main/java/io/druid/server/http/HistoricalResource.java]: If we only need the status code, how about `Response.ok().build()`?

File: server/src/main/java/io/druid/server/http/HistoricalResource.java [hunk 2]
```diff
@@ -52,4 +52,16 @@ public Response getLoadStatus()
   {
     return Response.ok(ImmutableMap.of("cacheInitialized", coordinator.isStarted())).build();
   }
+
+  @GET
+  @Path("/loadStatusCode")
```
Comment (xanec) [inline server/src/main/java/io/druid/server/http/HistoricalResource.java]: (Optional) How about using the endpoint `/readiness`. It's short, all small-caps, and relevant to its usage.

Comment (leventov) [reply server/src/main/java/io/druid/server/http/HistoricalResource.java]: Changed

File: server/src/main/java/io/druid/server/http/HistoricalResource.java (hunk 1 above)
Comment (leventov) [reply server/src/main/java/io/druid/server/http/HistoricalResource.java]: Changed

File: server/src/main/java/io/druid/server/http/HistoricalResource.java [hunk 3]
```diff
@@ -52,4 +52,15 @@ public Response getLoadStatus()
   {
     return Response.ok(ImmutableMap.of("cacheInitialized", coordinator.isStarted())).build();
   }
+
+  @GET
+  @Path("/readiness")
+  public Response getLoadStatusCode()
```
Comment (xanec) [inline server/src/main/java/io/druid/server/http/HistoricalResource.java]: Maybe we should change this to `getReadiness` for conformity?

File: docs/content/design/historical.md [hunk 4]
```diff
@@ -50,4 +50,11 @@ Returns the Druid version, loaded extensions, memory used, total memory and othe
...
 * `/druid/historical/v1/loadstatus`
 
-Returns a flag indicating if all segments in the local cache have been loaded. This can be used to know when a historical node is ready to be queried after a restart.
+Returns JSON of the form `{"cacheInitialized":<value>}`, where value is either `true` or `false` indicating if all
+segments in the local cache have been loaded. This can be used to know when a historical node is ready
+to be queried after a restart.
+
+* `/druid/historical/v1/readiness`
+
+Similar to `/druid/historical/v1/loadstatus`, but instead or returning JSON with a flag, responses 200 OK if segments
```
Comment (xanec) [inline docs/content/design/historical.md]: Typo here. Should be "...instead _**of**_ returning JSON with a flag, _**returns**_ 200 OK..."

File: server/src/main/java/io/druid/server/http/HistoricalResource.java (hunk 3 above)
Comment (leventov) [reply server/src/main/java/io/druid/server/http/HistoricalResource.java]: Sure, my bad of not changing it myself

File: docs/content/design/historical.md (hunk 4 above)
Comment (leventov) [reply docs/content/design/historical.md]: Fixed typo. I chose "responses" because in my perception it "returns" body, which is empty in this case

**Comment by himanshug** [comment]
what is the use case for this endpoint ?

**Comment by leventov** [comment]
@himanshug readiness check in [Marathon](https://mesosphere.github.io/marathon/)

**Pull request merged**